torch/serialization.py

```diff
@@ -625,7 +645,12 @@ def persistent_id(obj):
         zip_file.write_record(name, storage.data_ptr(), num_bytes)
 
 
-def load(f, map_location=None, pickle_module=pickle, **pickle_load_args):
+def load(
+    f: Union[str, os.PathLike, BinaryIO, IO[bytes]],
+    map_location: Optional[Union[Callable[[torch.Tensor, str], torch.Tensor], torch.device, str, Dict[str, str]]] = None,
```

Comment:
Very much not sure if this is correct, would appreciate a few more eyes on this.

Reply:
Looks correct based on the inline documentation, but again, I wonder if it would benefit from defining some subtypes, say `CallableMapLocation`